core/src/patterns/slice.rs

```diff
@@ -269,4 +260,21 @@ mod test {
         assert_eq!(empty.as_slice(), &[]);
         assert_eq!(slice, &[5, 6, 2, 3, 5]);
     }
+
+    #[test]
+    fn multi_borrow_mut_slice() {
+        let slice = &mut [0, 1, 2, 3, 5];
+        let empty = FFISliceMut::<u8>::empty();
+        let target: &mut [u8] = {
```

Comment:
Trying to borrow slice again results in a complier error, this is still a safe piece of code because the original slice is the scope of mutation, not the intermediate `FFISliceMut`.